linkerd/app/core/src/trace.rs

```diff
@@ -33,25 +45,52 @@ pub fn init_log_compat() -> Result<(), Error> {
     tracing_log::LogTracer::init().map_err(Error::from)
 }
 
-pub fn with_filter(filter: impl AsRef<str>) -> (Dispatch, LevelHandle) {
+pub fn with_filter_and_format(
+    filter: impl AsRef<str>,
+    format: impl AsRef<str>,
+) -> (Dispatch, LevelHandle) {
     let filter = filter.as_ref();
+    let format = format.as_ref();
 
     // Set up the subscriber
     let start_time = clock::now();
-    let builder = FmtSubscriber::builder()
-        .with_timer(Uptime { start_time })
-        .with_env_filter(filter)
-        .with_filter_reloading()
-        .with_ansi(cfg!(test));
-    let handle = LevelHandle {
-        inner: builder.reload_handle(),
-    };
-    let dispatch = Dispatch::new(builder.finish());
-
-    (dispatch, handle)
+
+    match format {
+        "JSON" => {
+            let builder = FmtSubscriber::builder()
+                .json()
+                .with_timer(Uptime { start_time })
+                .with_env_filter(filter)
+                .with_filter_reloading()
+                .with_ansi(cfg!(test));
```

Comment:
IIRC the `.with_ansi` option doesn't actually do anything with the JSON formatter --- it doesn't support ANSI colors.

Reply:
Thanks!